test/distributed/_tensor/test_optimizers.py

```diff
@@ -349,6 +349,45 @@ def test_adadelta_1d_sharding(self):
             inp = torch.ones(8, 10, device=self.device_type)
             self._assert_optimizer(mesh, mod, opt, dist_mod, dist_opt, inp)
 
+    @with_comms
+    def test_nadam_1d_sharding(self):
+        mesh = DeviceMesh(self.device_type, list(range(self.world_size)))
+
+        nadam_configs = [
+            {"lr": 0.1},
+            {"lr": 0.1, "weight_decay": 0.05},
+            {"lr": 0.1, "weight_decay": 0.05, "foreach": True},
+            {
+                "lr": 0.1,
+                "betas": (0.6, 0.66),
+                "eps": 1e-6,
+                "weight_decay": 0.05,
+                "foreach": True,
+            },
+            {
+                "lr": 0.1,
+                "betas": (0.6, 0.66),
+                "eps": 1e-6,
+                "weight_decay": 0.05,
+                "decoupled_weight_decay": True,
+                "foreach": True,
+            },
+        ]
+
+        for config in nadam_configs:
+            mod = MLPModule(self.device_type)
```

Comment:
we can refactor this common logic into a util function and reuse it in `test_optimizers.py`

Reply:
I'll see what I can do and do it in a follow up PR, I think for model/optim init we might want to have them explicit constructed in test cases so that it's easy to understand what this test is testing about